### ISSUE TITLE
split() function returns unexpected result in splitting empty string

### ISSUE THREAD

**Issue by RomTin** [hashicorp/terraform#19904]
### Terraform Version

```
Terraform v0.11.10
+ provider.aws v1.54.0
+ provider.null v1.0.0

Your version of Terraform is out of date! The latest version
is 0.11.11. You can update by downloading from www.terraform.io/downloads.html
```

### Terraform Configuration Files

```hcl
resource "null_resource" "dummy"{
}

locals {
  split_null_by_null = "${split("", "")}"
  split_foo_by_null = "${split("", "foo")}"
  split_foo_by_bar = "${split("bar", "foo")}"
  split_foo_by_foo = "${split("foo", "foo")}"
  split_null_by_foo = "${split("foo", "")}"
}

output "split_null_by_null" {
  value = "${local.split_null_by_null}"
}

output "len_split_null_by_null" {
  value = "${length(local.split_null_by_null)}"
}

output "split_foo_by_null" {
  value = "${local.split_foo_by_null}"
}

output "len_split_foo_by_null" {
  value = "${length(local.split_foo_by_null)}"
}

output "split_foo_by_bar" {
  value = "${local.split_foo_by_bar}"
}

output "len_split_foo_by_bar" {
  value = "${length(local.split_foo_by_bar)}"
}

output "split_foo_by_foo" {
  value = "${local.split_foo_by_foo}"
}

output "len_split_foo_by_foo" {
  value = "${length(local.split_foo_by_foo)}"
}

output "split_null_by_foo" {
  value = "${local.split_null_by_foo}"
}

output "len_split_null_by_foo" {
  value = "${length(local.split_null_by_foo)}"
}
```
### Actual Behavior

For the cases, `split_foo_by_bar`, `split_foo_by_foo`, `split_foo_by_null` and `split_null_by_foo` are understandable like below.

* `split_foo_by_bar`: `split("bar", "foo") -> ["foo"]`
* `split_foo_by_foo`: `split("foo", "" + "foo" + "") -> ["", ""]`
* `split_foo_by_null`: `split("", "f" + "" + "o" + "" + "o") -> ["f", "o", "o"]`
* `split_null_by_foo`: `split("foo", "") -> [""]`

Then, the result of `split_null_by_null` is a little bit incomprehensible. Why did `split()` return empty list? I thought `split("", "")` should behaves same as `split("foo", "foo")` did.

```
null_resource.dummy: Refreshing state... (ID: ..........)

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

len_split_foo_by_bar = 1
len_split_foo_by_foo = 2
len_split_foo_by_null = 3
len_split_null_by_foo = 1
len_split_null_by_null = 0
split_foo_by_bar = [
    foo
]
split_foo_by_foo = [
    ,

]
split_foo_by_null = [
    f,
    o,
    o
]
split_null_by_foo = [

]
split_null_by_null = []
```

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`
3. `terraform output` (Optional)

**Comment by llamahunter** [comment]
Also, split(",", join(",", list())) returns [ "" ], not [].

The documentation of split() says "Splits the string previously created by join back into a list".  Obviously does not obey this contract in this instance.

**Comment by robkinyon-roivant** [comment]
Apparently, this is considered "correct behavior" according to https://github.com/hashicorp/terraform/issues/12472#issuecomment-511616663. There's a few ways to address this inconsistency with nearly every other implementation of `split()` in existence, but it's dumb.

**Comment by mildwonkey** [comment]
The `split` function was entirely re-implemented for terraform v0.13, so I am going to close this issue as out of date. if you still experience a problem with split in terraform v0.13 and there isn't already an issue or feature request open, please open a new issue. Thanks!
